assets/js/interactions.ts

```diff
@@ -2,135 +2,165 @@
   cacheStatus(imageId, 'voted', 'up');
   onImage(imageId, '.interaction--upvote', el => el.classList.add('active'));
 }
 
-function showDownvoted(imageId) {
+function showDownvoted(imageId: string) {
   cacheStatus(imageId, 'voted', 'down');
   onImage(imageId, '.interaction--downvote', el => el.classList.add('active'));
 }
 
-function showFaved(imageId) {
+function showFaved(imageId: string) {
   cacheStatus(imageId, 'faved', '');
   onImage(imageId, '.interaction--fave', el => el.classList.add('active'));
 }
 
-function showHidden(imageId) {
+function showHidden(imageId: string) {
   cacheStatus(imageId, 'hidden', '');
   onImage(imageId, '.interaction--hide', el => el.classList.add('active'));
 }
 
-function resetVoted(imageId) {
+function resetVoted(imageId: string) {
   uncacheStatus(imageId, 'voted');
   onImage(imageId, '.interaction--upvote', el => el.classList.remove('active'));
   onImage(imageId, '.interaction--downvote', el => el.classList.remove('active'));
 }
 
-function resetFaved(imageId) {
+function resetFaved(imageId: string) {
   uncacheStatus(imageId, 'faved');
   onImage(imageId, '.interaction--fave', el => el.classList.remove('active'));
 }
 
-function resetHidden(imageId) {
+function resetHidden(imageId: string) {
   uncacheStatus(imageId, 'hidden');
   onImage(imageId, '.interaction--hide', el => el.classList.remove('active'));
 }
 
-function interact(type, imageId, method, data = {}) {
+function interact(type: EndpointType, imageId: string, method: HttpMethod, data: Record<string, unknown> = {}) {
   return fetchJson(method, endpoints[type](imageId), data)
     .then(res => res.json())
-    .then(res => setScore(imageId, res));
+    .then((res: ScorePayload) => setScore(imageId, res));
 }
 
-function displayInteractionSet(interactions) {
+function displayInteractionSet(interactions: (Interaction | CacheRecord)[]) {
   interactions.forEach(i => {
-    switch (i.interaction_type) {
-      case 'faved':
-        showFaved(i.image_id);
-        break;
-      case 'hidden':
-        showHidden(i.image_id);
-        break;
-      default:
-        if (i.value === 'up') showUpvoted(i.image_id);
-        if (i.value === 'down') showDownvoted(i.image_id);
+    if ('image_id' in i) {
+      const imageId = String(i.image_id);
+      switch (i.interaction_type) {
+        case 'faved':
+          showFaved(imageId);
+          break;
+        case 'hidden':
+          showHidden(imageId);
+          break;
+        default:
+          if (i.value === 'up') showUpvoted(imageId);
+          if (i.value === 'down') showDownvoted(imageId);
+      }
+    } else {
+      const imageId = String(i.imageId);
+      switch (i.interactionType) {
+        case 'faved':
+          showFaved(imageId);
+          break;
+        case 'hidden':
+          showHidden(imageId);
+          break;
+        default:
+          if (i.value === 'up') showUpvoted(imageId);
+          if (i.value === 'down') showDownvoted(imageId);
+      }
```

Comment:
these cases are straight up copy and pasted, make a separate function that just takes the image id and have this one call it